src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -560,6 +573,10 @@ void TableDockWidget::showAllGroups() {
     if (viewType == groupView)
       setIntensityColName();
     setVisible(false);
+    _mainwindow->ligandWidget->resetColor();
+    _mainwindow->removePeaksTable(this);
+    noPeakFound();
+    this->~TableDockWidget();
```

Comment:
Don't do this. Since table deletion makes multiple other calls, simply calling the destructor might block the event loop. When deleting widget (or any `QObject`), always call the `deleteLater` method on the instance. It handles the deallocation asynchronously.

Side note: whenever you do have to call destructors, just use `delete object` and not `this->~ObjectClass()`.